packages/server/src/service-worker-manager.ts

```diff
@@ -91,28 +123,69 @@ export class ServiceWorkerManager implements IServiceWorkerManager {
 
     this._setRegistration(registration);
 
+    // transfer the port for communication with the Service Worker
+    void serviceWorker.controller?.postMessage(
+      {
+        type: 'INIT_PORT',
+      },
+      [this._messageChannel.port2],
```

Comment:
We're sending a new `INIT_PORT` message for each tab, right? But the service worker has one global `messagePort` variable shared by all tabs? Won't thing bring an issue?

Reply:
👍 need a tab_uuid here

Reply:
and a `beforeunload` handler to remove the port associated with the current tab uuid